server/src/main/java/org/elasticsearch/upgrades/SystemIndexMigrator.java

```diff
@@ -381,6 +383,41 @@ private void migrateSingleIndex(ClusterState clusterState, Consumer<BulkByScroll
         }
         Index oldIndex = imd.getIndex();
         String newIndexName = migrationInfo.getNextIndexName();
+
+        final String v2template = MetadataIndexTemplateService.findV2Template(clusterState.metadata(), newIndexName, false);
```

Comment:
I thought v2 templates already don't match system indices, and the problem was only caused by legacy templates - I'll try to double check this, but I'll also ask: Am I just straight-up wrong about that?

Reply:
I think both apply, is this legacy or not template usage from kibana: https://github.com/elastic/kibana/blob/3730dd0779ed8efd74aee88f57422781ec1ac122/x-pack/plugins/security/server/session_management/session_index.ts#L61-L92